Add customization options section (#127)
bump-sh/docs: src/_help/customization-options/operations-navigation.md

---
title: Operations and navigation
---

- TOC
{:toc}

Bump.sh offers several methods for sorting your API operations to provide a smooth and coherent reading structure, based on either the tags or operation names and URLs.

Supported API specifications allow for several types of requests:

- Endpoints are typically URLs that point to a specific resource on an API server. In OpenAPI, endpoints are stored under the `paths` resource.

> - Webhooks are requests generated by the API and sent to an external server after a specific event has occurred. In OpenAPI 3.1, they are stored under the `webhooks` resource.

Reviewer: I don't get why we talk about that here?

> > If first-level field `tags` are present at the [root of your OpenAPI document object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#openapi-object), Bump.sh will use `Group by tags` as a default documentation generation behaviour.

Reviewer: This should be moved in a `## Automatically` section, explaining the default (= if "Automatically" is selected) mode of grouping.

---
title: Operations and navigation
---

- TOC
{:toc}

> Bump.sh offers several methods for sorting your API operations to provide a smooth and coherent reading structure, based on either the tags or the endpoints of your API definition.

Reviewer: ```suggestion
Bump.sh offers several methods for sorting your API operations to provide a smooth and coherent reading structure, based on either the tags or the operation names and URLs.
```

WDYT of this change? It seems a bit more clear to me here.

Author: Way much better. I suppose that I was just proud to correctly use "API definition" :D

Reviewer: Haha, and it was perfectly used ;)

The OpenAPI specification allows for several types of requests:

- Endpoints are typically URLs that point to a specific resource on an API server. In OpenAPI, endpoints are stored under the `paths` resource.

> - Webhooks are requests generated by the API and sent to an external server after a specific event has occurred. In OpenAPI 3.1, they are stored under the `webhooks` resource.

Reviewer: We shouldn't go specific with OpenAPI here, it's also valid for AsyncAPI. We need to stay generic.

> Bump.sh can use its resources to group the display of operations in your documentation. You can also use [tags](https://spec.openapis.org/oas/v3.1.0#tag-object), which are used, among other things, to group operations by resource and should be added to your API definition.

Reviewer 2: ```suggestion
Bump.sh can use these resources to group the display of operations in your documentation. You can also use [tags](https://spec.openapis.org/oas/v3.1.0#tag-object), which are used, among other things, to group operations by resource. Tags must be defined in your API definition.
```

> Bump.sh can use these resources to group the display of operations in your documentation. You can also use [tags](https://spec.openapis.org/oas/v3.1.0#tag-object), which are used, among other things, to group operations by resource. Tags must be defined in your API definition.

Reviewer: I don't find this sentence useful here, as we describe later what are paths and tags. Can we remove it?

![](/images/help/operations.png)

> ## Group operations by path (endpoints)

Reviewer: We shouldn't talk about endpoints here, as it's a REST only concept. Let's talk about "Automatic grouping by operation details" or "Automatic grouping by path" or something like that

> When group operations by path is chosen, Bump.sh deduce endpoint name from related path. First part of the path is extracted to generate the endpoint name, and every operation related to this endpoint name will be grouped together.

Reviewer 2: ```suggestion
When `Group operations by path` is chosen, Bump.sh deduces endpoint names from related paths. The first part of the path is extracted to generate the endpoint name, and every operation related to this endpoint name will be grouped together.
```

Reviewer 2: Not sure how to best present this:
"Group operations by path" is reader-friendlier
`Group by path` is UI-exact
Still, I recommend using one of those as "When group operations by path is chosen" is somewhat hard to read otherwise, from a syntax perspective.

Author: Won't stick to UI perfectly as it should be more user-friendly : it doesn't block the reader to find the option in the dashboard.
Not sure of using code blocks for something else than code blocks (this is a personal introspection)

> When `Group operations by path`` is chosen, Bump.sh deduces endpoint names from related paths. The first part of the path is extracted to generate the endpoint name, and every operation related to this endpoint name will be grouped together.

Reviewer: ```suggestion
When `Group operations by path`` is chosen, Bump.sh deduces group names from related paths. The first part of the path is extracted to generate the group name, and every operation related to it will be grouped together.
```

![](/images/help/group-by-path.png)

## Group operations by tag

> > If first level field `tags` are present at the [root of your OpenAPI document object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#openapi-object), Bump.sh will use `Group by tags` as a default documentation generation behaviour.

Reviewer 3: ```suggestion
> If first-level field `tags` are present at the [root of your OpenAPI document object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#openapi-object), Bump.sh will use `Group by tags` as a default documentation generation behaviour.
```

> > If first-level field `tags` are present at the [root of your OpenAPI document object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#openapi-object), Bump.sh will use `Group by tags` as a default documentation generation behaviour.

Reviewer: We need to add a word about the fact that it's only if the setting is set to "default". This should be moved to the section's introduction.

{: .info}

> When group operation by tag is selected, Bump.sh will use the [`tags`](https://spec.openapis.org/oas/v3.1.0#tag-object) from your API definition to group, name and sort operations.

Reviewer: ```suggestion
When "Group operation by tag" is selected, Bump.sh will use the [`tags`](https://spec.openapis.org/oas/v3.1.0#tag-object) from your API definition to group, name and sort operations.
```

> Tags offer you better customization of your documentation, going beyond the resource names. You can also add a description for each tag, which will appear in the header of your documentation.

Reviewer 3: Would it be possible to explain in which way it's better?

Author: Not sure of what would be a great explanation here, I mean, in addition of what's already written.
